tests/unit/utils/test_urlutils.py

```diff
@@ -277,26 +277,40 @@ def test_get_path_unicode_encode_error(self, os_mock, caplog):
 def test_special_urls(url, special):
     assert urlutils.is_special_url(QUrl(url)) == special
 
-
-@pytest.mark.parametrize('url, host, query', [
-    ('testfoo', 'www.example.com', 'q=testfoo'),
-    ('test testfoo', 'www.qutebrowser.org', 'q=testfoo'),
-    ('test testfoo bar foo', 'www.qutebrowser.org', 'q=testfoo bar foo'),
-    ('test testfoo ', 'www.qutebrowser.org', 'q=testfoo'),
-    ('!python testfoo', 'www.example.com', 'q=%21python testfoo'),
-    ('blub testfoo', 'www.example.com', 'q=blub testfoo'),
-    ('stripped ', 'www.example.com', 'q=stripped'),
-    ('test-with-dash testfoo', 'www.example.org', 'q=testfoo'),
+@pytest.mark.parametrize('url, host, query, open_base_url', [
+    ('testfoo', 'www.example.com', 'q=testfoo', False),
+    ('test testfoo', 'www.qutebrowser.org', 'q=testfoo', False),
+    ('test testfoo bar foo', 'www.qutebrowser.org', 'q=testfoo bar foo', False),
+    ('test testfoo ', 'www.qutebrowser.org', 'q=testfoo', False),
+    ('!python testfoo', 'www.example.com', 'q=%21python testfoo', False),
+    ('blub testfoo', 'www.example.com', 'q=blub testfoo', False),
+    ('stripped ', 'www.example.com', 'q=stripped', False),
+    ('test-with-dash testfoo', 'www.example.org', 'q=testfoo', False),
+    ('testfoo', 'www.example.com', 'q=testfoo', True),
+    ('test testfoo', 'www.qutebrowser.org', 'q=testfoo', True),
+    ('test testfoo bar foo', 'www.qutebrowser.org', 'q=testfoo bar foo', True),
+    ('test testfoo ', 'www.qutebrowser.org', 'q=testfoo', True),
+    ('!python testfoo', 'www.example.com', 'q=%21python testfoo', True),
+    ('blub testfoo', 'www.example.com', 'q=blub testfoo', True),
+    ('stripped ', 'www.example.com', 'q=stripped', True),
+    ('test-with-dash testfoo', 'www.example.org', 'q=testfoo', True),
```

Comment:
That's a lot of duplication - I'd recommend instead having a second `@pytest.mark.parametrize('open_base_url', [True, False])` (which will then test all combinations), and having a second test for `get_search_url` for the last two cases (with `open_base_url = True` and no query).